### ISSUE TITLE
Semantic Segmentation Backbone Model Training

### ISSUE THREAD

**Issue by dlee640** [matej-ulicny/harmonic-networks#3]
Hello. Thank you for open sourcing an inspiring work.

I have come to find this repo from reading the recent paper on "Harmonic CNN based on DCT". In section 5.3 Semantic Segmentation on Pascal VOC, there was a content regarding "We replace the backbone model of the segmentation network with harmonic ResNets using two settings for the backbone" and latter approach was "taking a harmonic models pre-trained on ImageNet". 

We are currently working to improve the accuracy of [ESANet](https://github.com/TUI-NICR/ESANet) for indoor semantic segmentation. ESANet utilizes RGB-D image, and we wanted to add separate channel that constitutes coefficients from DCT. In order to fuse this DCT channel into RGB-D channels, we need to retrain ResNet backbone using this harmonic network. 

_**Can you provide some insight / steps on how I can utilize repo to obtain harmonic backbones pretrained on ImageNet? Thank you.**_

**Comment by matej-ulicny** [comment]
Thank you for taking interest in this work. You can find definitions for harmonic ResNets [here](https://github.com/matej-ulicny/harmonic-networks/blob/master/imagenet/models/resnet.py). 
After installing the package locally (as described in the readme file) in 'imagenet' folder you can run
```
python main.py -a resnet101 ./path/to/imagenet/ --batch-size 256 --harm_root --harm_res_blocks --evaluate --pretrained
```
to evaluate the ResNet101 on ImageNet. Using option '--pretrained' automatically downloads the model. Alternatively you can find link to the checkpoint in the [code](https://github.com/matej-ulicny/harmonic-networks/blob/master/imagenet/models/resnet.py). 

If you intend to incorporate this backbone inside another model you can adjust the ResNet definition above to be compatible with your code. Note that the models have been trained only on RGB channels and the DCT is calculated inside the layers, you might need to make modifications to the first layer for the purpose you have described.
